src/Block/Fetch.hs

```diff
@@ -119,26 +119,29 @@ receiveLoop ::
     , MonadUnliftIO m
     , MonadLogger m
     , MonadReader r m
-    , Has DatumFilter r
     , Has Hasql.Connection r
     ) =>
     WS.Connection ->
+    DatumFilter ->
     m ()
-receiveLoop conn = do
+receiveLoop conn datumFilter = do
     jsonMsg <- liftIO $ WS.receiveData conn
     let msg = Json.decode @OgmiosFindIntersectResponse jsonMsg
     case _result <$> msg of
         Nothing -> do
             logErrorNS "receiveLoop" "Error decoding FindIntersect response"
         Just (IntersectionNotFound _) -> do
-            logErrorNS "receiveLoop" "Find intersection error: Intersection not found"
+            logErrorNS "receiveLoop" "Find intersection error: Intersection not found. Consider restarting block fetcher with different block info"
         Just (IntersectionFound _ _) -> do
             logInfoNS "receiveLoop" "Find intersection: intersection found, starting RequestNext loop"
-            Async.withAsync (receiveBlocksLoop conn) $ \receiveBlocksWorker -> do
+            Async.withAsync (receiveBlocksLoop conn datumFilter) $ \receiveBlocksWorker -> do
```

Comment:
`datumFilter` is unused here

Reply:
It is, it's passed to `receiveBlocksLoop` then to `saveDatumsFromAlonzoBlock` and used to filter datums